0000-api-versions.md

```diff
@@ -0,0 +1,43 @@
+# Meta
+[meta]: #meta
+- Name: API Versions
+- Start Date: 2019-08-05
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+Begin versioning the various APIs that the spec describes. Then require these versions be specified in related TOML files, under an `api` field.
+
+# Motivation
+[motivation]: #motivation
+
+To enable logic and/or validation based on `api`. For instance, we could keep users from attempting to use a combination of incompatible buildpacks and lifecycle.
+
+# What it is
+[what-it-is]: #what-it-is
+
+There would be two pieces to this:
+
+1. A reference to the current versions of all APIs placed somewhere in the spec (location within spec TBA). For instance, a Buildpack API and a Platform API.
```

Comment:
would this mean there's a distribution API (thinking that every file in the spec repo is a API version then?)

Reply:
Potentially, though I'm not 100% sure how we want to break things down into APIs (by high-level sub-spec could be one option).